system-test/cluster.ts

```diff
@@ -0,0 +1,326 @@
+            });
+            assert.fail();
+          } catch (e) {
+            assert.equal(e.message, ClusterUtils.noConfigError);
+          }
+        });
+        it('should throw an error when providing manual and autoscaling configurations', async () => {
+          try {
+            await cluster2.create({
+              location: 'us-west1-c',
+              nodes: 2,
+              minServeNodes: 3,
+            });
+            assert.fail();
+          } catch (e) {
+            assert.equal(e.message, ClusterUtils.allConfigError);
+          }
+        });
+        it('should throw an error providing all autoscaling configurations', async () => {
+          try {
+            await cluster2.create({
+              location: 'us-west1-c',
+              minServeNodes: 3,
+              cpuUtilizationPercent: 51,
+            });
+            assert.fail();
+          } catch (e) {
+            assert.equal(e.message, ClusterUtils.incompleteConfigError);
+          }
+        });
+      });
+    });
+    describe('With automatic scaling', () => {
+      const minServeNodes = 2;
+      const maxServeNodes = 4;
+      const cpuUtilizationPercent = 50;
+
+      const createClusterOptions = {
+        location: 'us-west1-c',
+        minServeNodes,
+        maxServeNodes,
+        cpuUtilizationPercent,
+      };
+      it('should create an instance with clusters for automatic scaling', async () => {
+        const clusterId = generateId('cluster');
+        await createNewInstance([
+          Object.assign({id: clusterId}, createClusterOptions),
+        ]);
+        const cluster: Cluster = instance.cluster(clusterId);
+        await checkMetadata(
+          cluster,
+          {
+            ...createClusterOptions,
+            nodes: minServeNodes,
+          },
+          true
+        );
+      });
+      it('should create an instance and then create clusters for automatic scaling', async () => {
+        const clusterId: string = generateId('cluster');
+        await createStandardNewInstance(clusterId, 2);
+        const clusterId2: string = generateId('cluster');
+        const cluster: Cluster = instance.cluster(clusterId2);
+        await cluster.create(createClusterOptions);
+        await checkMetadata(
+          cluster,
+          {
+            ...createClusterOptions,
+            nodes: minServeNodes,
+          },
+          true
+        );
+      });
+    });
+  });
+  describe('Update cluster', () => {
+    describe('Starting from manual scaling', () => {
+      let cluster: Cluster;
+      const startingNodes = 2;
+
+      beforeEach(async () => {
+        const clusterId = generateId('cluster');
+        await createStandardNewInstance(clusterId, startingNodes);
+        cluster = instance.cluster(clusterId);
+      });
+
+      it('should change nodes for manual scaling', async () => {
+        const updateNodes = 5;
+        assert.notEqual(startingNodes, updateNodes);
```

Comment:
this check is comparing two int variables - if you want to have this, I would instead get the creater cluster's nodes, then compare the created cluster's nodes to the updateNodes value.

Reply:
`checkMetadata` does the second thing you mentioned though it checks `metadata.serveNodes` equals `updateNodes` rather than checking `metadata.serveNodes` does not equal `startingNodes`. This `assert.notEqual` is more of a maintainability thing to ensure the programmer doesn't set `updateNodes` to equal `startingNodes` and then have the test pass when `setMetadata` doesn't actually work properly. If you think it's confusing though then we can remove it.

Reply:
I don't think this is providing much in terms of autoscaling assertions, so I would remove it in favor of the checks you already mentioned.